scilpy/tractograms/streamline_and_mask_operations.py

```diff
@@ -100,42 +118,24 @@ def cut_outside_of_mask_streamlines(sft, binary_mask, min_len=0):
     sft.to_corner()
     streamlines = sft.streamlines
 
-    new_streamlines = []
-    for _, streamline in enumerate(streamlines):
-        entry_found = False
-        last_success = 0
-        curr_len = 0
-        longest_seq = (0, 0)
-        for ind, pos in enumerate(streamline):
-            pos = tuple(pos.astype(np.int16))
-            if binary_mask[pos]:
-                if not entry_found:
-                    entry_found = True
-                    last_success = ind
-                    curr_len = 0
-                else:
-                    curr_len += 1
-                    if curr_len > longest_seq[1] - longest_seq[0]:
-                        longest_seq = (last_success, ind + 1)
-            else:
-                if entry_found:
-                    entry_found = False
-                    if curr_len > longest_seq[1] - longest_seq[0]:
-                        longest_seq = (last_success, ind)
-                        curr_len = 0
-        # print(longest_seq)
-        if longest_seq[1] != 0:
-            new_streamlines.append(streamline[longest_seq[0]:longest_seq[1]])
+    # Cut streamlines within the mask and return the new streamlines
+    # New endpoints may be generated
+    new_streamlines = _cut_streamlines_with_masks(
+        streamlines, binary_mask, binary_mask)
 
     new_sft = StatefulTractogram.from_sft(new_streamlines, sft)
     return filter_streamlines_by_length(new_sft, min_length=min_len)
 
 
 def cut_between_masks_streamlines(sft, binary_mask, min_len=0):
```

Comment:
I find the name confusing. When reading the unit tests, I had to come back here to understand why it says between_masks but we only give one mask. 

Could it be between_two_mask_blobs or I don't know..... something more explicit?

Reply:
Changed the name, still a bit clunky but should be clearer now.